runtime/compiler/build/toolcfg/gnu/common.mk

```diff
@@ -138,10 +138,6 @@ ifeq ($(HOST_ARCH),x)
         CX_DEFINES+=J9HAMMER
         CX_FLAGS+=-m64 -fPIC
     endif
-
-    ifneq ($(JITSERVER_SUPPORT),)
```

Comment:
Don't we need these defines? Are the definitions added in some other place?

Reply:
This check is for defining the macro `JITSERVER_SUPPORT`. We should instead use `J9VM_OPT_JITSERVER` which gets defined in `j9cfg.h` (when using UMA build system) or by `CMakeLists.txt` in VM and JIT.
Normally `j9cfg.h` gets included via `j9.h` and any code using `J9VM_OPT_JITSERVER` should now include `j9.h`.
Before this PR, JIT files didn't include `j9.h` if there isn't any dependency on it, but with this PR if the JIT files need to see `J9VM_OPT_JITSERVER`, then `j9.h` needs to be included.
A case in point is https://github.com/eclipse/openj9/pull/8656/files#diff-65a548d2a6dcefcbcdd234568cac76ab where I have to explicitly include `j9.h`.